client/src/App.js

```diff
@@ -2,39 +2,70 @@ import React, { useState, useEffect } from 'react';
 import './App.css';
 import Dashboard from './Components/Dashboard/Dashboard';
 import LoginPage from './Components/LoginPage/LoginPage';
+import PrivateRoute from './Components/PrivateRoute/PrivateRoute';
+import {
+  BrowserRouter as Switch,
+  Route,
+  BrowserRouter,
+  Redirect,
+} from 'react-router-dom';
+import { getIdFromLocation } from './helperFunc';
 
 function App() {
-  const [token, setToken] = useState('');
-  const [username, setUsername] = useState('');
-  const [offline, setOffline] = useState(!navigator.onLine);
+  const [token, setToken] = useState(localStorage.getItem('token') || '');
+  const [username, setUsername] = useState(
+    localStorage.getItem('username') || ''
+  );
+
+  /* the user can access the app through two different options:
+  1. indicating username and token manually. In the case of GitHub Enterprise the user also needs to indicate the endpoint.
+  2. Github authentication */
 
   useEffect(() => {
-    if (localStorage.getItem('token')) {
-      setToken(localStorage.getItem('token'));
-      setUsername(localStorage.getItem('username'));
+    // in case that the user is authenticated through github, an id will be attached to the url.
+    const userId = getIdFromLocation();
+    if (userId) {
+      // if there's id, we need to look for the token stored in the ddbb
+      fetch(`http://localhost:8080/users/${userId}`)
```

Comment:
Have you thought about using environment variables to set the API_URL? I think it could be a good step to upload your code without needing to app another PR to change this :) 
```suggestion
      fetch(`${process.env.API_URL}/users/${userId}`)
```